permission/permission.go

```diff
@@ -0,0 +1,759 @@
+	var evtNodeActivated *pbind.NodeManagerNodeActivated
+	var evtNodeBlacklisted *pbind.NodeManagerNodeBlacklisted
+
+	opts := &bind.WatchOpts{}
+	var blockNumber uint64 = 1
+	opts.Start = &blockNumber
+
+	if _, err := p.permNode.NodeManagerFilterer.WatchNodeApproved(opts, chNodeApproved); err != nil {
+		log.Info("Failed WatchNodeApproved", "error", err)
+	}
+
+	if _, err := p.permNode.NodeManagerFilterer.WatchNodeProposed(opts, chNodeProposed); err != nil {
+		log.Info("Failed WatchNodeProposed", "error", err)
+	}
+
+	if _, err := p.permNode.NodeManagerFilterer.WatchNodeDeactivated(opts, chNodeDeactivated); err != nil {
+		log.Info("Failed NodeDeactivated", "error", err)
+	}
+	if _, err := p.permNode.NodeManagerFilterer.WatchNodeActivated(opts, chNodeActivated); err != nil {
+		log.Info("Failed WatchNodeActivated", "error", err)
+	}
+
+	if _, err := p.permNode.NodeManagerFilterer.WatchNodeBlacklisted(opts, chNodeBlacklisted); err != nil {
+		log.Info("Failed NodeBlacklisting", "error", err)
+	}
+
+	for {
+		select {
+		case evtNodeApproved = <-chNodeApproved:
+			p.updatePermissionedNodes(evtNodeApproved.EnodeId, NodeAdd)
+			types.NodeInfoMap.UpsertNode(evtNodeApproved.OrgId, evtNodeApproved.EnodeId, types.NodeApproved)
+
+		case evtNodeProposed = <-chNodeProposed:
+			types.NodeInfoMap.UpsertNode(evtNodeProposed.OrgId, evtNodeProposed.EnodeId, types.NodePendingApproval)
+
+		case evtNodeDeactivated = <-chNodeDeactivated:
+			p.updatePermissionedNodes(evtNodeDeactivated.EnodeId, NodeDelete)
+			types.NodeInfoMap.UpsertNode(evtNodeDeactivated.OrgId, evtNodeDeactivated.EnodeId, types.NodeDeactivated)
+
+		case evtNodeActivated = <-chNodeActivated:
+			p.updatePermissionedNodes(evtNodeActivated.EnodeId, NodeAdd)
+			types.NodeInfoMap.UpsertNode(evtNodeActivated.OrgId, evtNodeActivated.EnodeId, types.NodeApproved)
+
+		case evtNodeBlacklisted = <-chNodeBlacklisted:
+			p.updatePermissionedNodes(evtNodeBlacklisted.EnodeId, NodeDelete)
+			p.updateDisallowedNodes(evtNodeBlacklisted.EnodeId)
+			types.NodeInfoMap.UpsertNode(evtNodeBlacklisted.OrgId, evtNodeBlacklisted.EnodeId, types.NodeBlackListed)
+		}
+	}
+}
+
+// updates node information in the permissioned-nodes.json file based on node
+// management activities in smart contract
+func (p *PermissionCtrl) updatePermissionedNodes(enodeId string, operation NodeOperation) {
+	log.Debug("updatePermissionedNodes", "DataDir", p.dataDir, "file", params.PERMISSIONED_CONFIG)
+
+	path := filepath.Join(p.dataDir, params.PERMISSIONED_CONFIG)
+	if _, err := os.Stat(path); err != nil {
+		log.Error("Read Error for permissioned-nodes.json file. This is because 'permissioned' flag is specified but no permissioned-nodes.json file is present", "err", err)
+		return
+	}
+	// Load the nodes from the config file
+	blob, err := ioutil.ReadFile(path)
+	if err != nil {
+		log.Error("updatePermissionedNodes: Failed to access permissioned-nodes.json", "err", err)
+		return
+	}
+
+	nodelist := []string{}
```

Comment:
consider using `var nodelist []string` instead of literal

Reply:
changes done.